### PR TITLE
docs: how delta lake transactions work

### PR DESCRIPTION
**Pull request by MrPowers**
@nkarpov and I collaborated on this Delta Lake transactions post.

It's meant to give the basics on how transactions work and why they're a huge advantage of Delta Lakes.

@rtyler is giving a talk on transactions/concurrency soon.  We're trying to set the stage with some foundational content first.

### CODE REVIEW

File: docs/how-delta-lake-works/delta-lake-acid-transactions.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+* If a computation cluster that’s writing the Delta table crashes for some reason
+* Two operations are running concurrently and one of them fails
+
+Successful transactions are always registered in the Delta table and persisted no matter what.
+
+## Conclusion
+
+Delta Lake supports transactions which provide necessary reliability guarantees for production data systems.
+
+Data lakes don’t provide transactions and this can cause nasty bugs and a bad user experience.  Let’s look at a couple of scenarios when the lack of transactions cause a poor user experience:
```
Comment (ion-elgreco) [inline docs/how-delta-lake-works/delta-lake-acid-transactions.md:263]: Maybe call it vanilla data lakes? Delta lake in my opinion is just a data lake with a  metadata layer

File: docs/how-delta-lake-works/delta-lake-acid-transactions.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+Delta Lake uses concurrency control to ensure that transactions are executed sequentially, even when user operations are performed concurrently.  The next page of this guide explains concurrency in Delta Lake in detail.
+
+**Durable transactions**
+
+Delta tables are generally persisted in cloud object stores which provide durability guarantees.
+
+Durability means that all transactions that are successfully completed will always remain persisted, even if there are service outages or program crashes.
+
+Suppose you have a Delta table that’s persisted in Azure blog storage.  The Delta table transactions that are committed will always remain available, even in these circumstances:
```
Comment (djouallah) [inline docs/how-delta-lake-works/delta-lake-acid-transactions.md:251]: blog - blob

File: docs/how-delta-lake-works/delta-lake-acid-transactions.md (hunk 1 above)
Comment (MrPowers) [reply docs/how-delta-lake-works/delta-lake-acid-transactions.md:263]: Updated this.  Delta Lakes def aren't data lakes 😱 

See [the Lakehouse paper](https://www.cidrdb.org/cidr2021/papers/cidr2021_paper17.pdf).  We usually call these "Lakehouse storage systems" or "open table formats".

File: docs/how-delta-lake-works/delta-lake-acid-transactions.md (hunk 2 above)
Comment (MrPowers) [reply docs/how-delta-lake-works/delta-lake-acid-transactions.md:251]: good catch, lol.  Thanks for reviewing!!

**Review by ion-elgreco — APPROVED** [review]
Thanks @MrPowers!

**Pull request merged**